cmd/new.go

```diff
@@ -402,6 +379,76 @@ func installDependencies(message string) error {
 	return nil
 }
 
+// runUpOrPrintNextSteps runs `up` automatically, or if `up` shouldn't run, prints out a message with next steps.
+func runUpOrPrintNextSteps(
+	stack backend.Stack, originalCwd string, cwd string, displayOpts backend.DisplayOptions, yes bool) error {
+
+	proj, root, err := readProject()
+	if err != nil {
+		return err
+	}
+
+	// Currently go projects require a build/install step before deployment, so we won't automatically run `up` for
+	// such projects. Once we switch over to using `go run` for go, we can remove this and always run `up`.
+	runUp := !strings.EqualFold(proj.RuntimeInfo.Name(), "go")
+
+	if runUp {
+		opts, err := updateFlagsToOptions(true /*interactive*/, false /*skipPreview*/, yes)
+		if err != nil {
+			return err
+		}
+		opts.Display = displayOpts
+		opts.Engine = engine.UpdateOptions{
+			Parallel: defaultParallel,
+		}
+
+		m, err := getUpdateMetadata("", root)
+		if err != nil {
+			return errors.Wrap(err, "gathering environment metadata")
+		}
+
+		_, err = stack.Update(commandContext(), proj, root, m, opts, cancellationScopes)
+		switch {
+		case err == context.Canceled:
+			return errors.New("update cancelled")
+		case err != nil:
+			return PrintEngineError(err)
+		default:
+			return nil
+		}
+	} else {
+		// If the current working directory changed, add instructions to cd into the directory.
+		var deployMsg string
+		if originalCwd != cwd {
+			// If we can determine a relative path, use that, otherwise use the full path.
+			var cd string
+			if rel, err := filepath.Rel(originalCwd, cwd); err == nil {
+				cd = rel
+			} else {
+				cd = cwd
+			}
+
+			// Surround the path with double quotes if it contains whitespace.
+			if containsWhiteSpace(cd) {
+				cd = fmt.Sprintf("\"%s\"", cd)
+			}
+
+			cd = fmt.Sprintf("cd %s", cd)
+
+			deployMsg = "To deploy it, '" + cd + "' and then run 'pulumi up'."
+			deployMsg = colors.Highlight(deployMsg, cd, colors.BrightBlue+colors.Underline+colors.Bold)
+		} else {
+			deployMsg = "To deploy it, run 'pulumi up'."
```

Comment:
It's a little weird to be showing this message for go templates that can't use `pulumi up` immediately, but this is the same message we show today. Would like to clean up